### PR TITLE
feat: CHP-179: Refactor fwoa archiving logic to not require transacti…

### PR DESCRIPTION
**Pull request by lquanx**
…on processing

Issue #, if available:

Description of changes:

Checklist:

* [ ] Have you successfully deployed to an AWS account with your changes?
* [ ] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: scripts/install.sh [hunk 1]
```diff
@@ -28,6 +28,7 @@ function usage(){
     echo "    --apigatewayClientErrorThreshold: API gateway 4xxerror threshold (Default: 5)"
     echo "    --lambdaErrorThreshold: lambda error latency threshold (Default: 1)"
     echo "    --ddbToESLambdaErrorThreshold: DDBToES lambda error threshold (Default: 1)"
+    echo "    --archiveConfig: archive config for specific resources. Format: resource1,ttl|resource2,ttl (Default: AuditEvent,15780000)"
```
Comment (lwarren01) [inline scripts/install.sh]: Default documented here doesn't match actual default.

File: serverless.yaml [hunk 2]
```diff
@@ -17,6 +17,7 @@ custom:
   apigatewayClientErrorThreshold: ${env:apigatewayClientErrorThreshold}
   lambdaErrorThreshold: ${env:lambdaErrorThreshold}
   ddbToESLambdaErrorThreshold: ${env:ddbToESLambdaErrorThreshold}
+  archiveConfig: ${env:archiveConfig, ''}
```
Comment (lwarren01) [inline serverless.yaml]: I know there are other existing env vars set here, but should the naming convention for env var names be ALL_CAPS? Also, should be push the default into here?

File: scripts/install.sh [hunk 3]
```diff
@@ -230,7 +232,10 @@ while [ "$1" != "" ]; do
                                                     ;;
         --ddbToESLambdaErrorThreshold )             shift
                                                     ddbToESLambdaErrorThreshold=$1
-                                                    ;;                                        
+                                                    ;;
+        --archiveConfig )                           shift
```
Comment (lwarren01) [inline scripts/install.sh]: Should we just expose this through to the serverless command as an env var, and avoid the silliness of defining a command option here?

File: scripts/install.sh [hunk 4]
```diff
@@ -232,10 +230,7 @@ while [ "$1" != "" ]; do
                                                     ;;
         --ddbToESLambdaErrorThreshold )             shift
                                                     ddbToESLambdaErrorThreshold=$1
-                                                    ;;
-        --archiveConfig )                           shift
-                                                    archiveConfig=$1
-                                                    ;;                                     
+                                                    ;;                                    
         -h | --help )                               usage
```
Comment (lquanx) [inline scripts/install.sh:234]: Lee, I have removed this parameter. archiveConfig can be passed as environment variables. like
`archiveConfig="Patient,3000" ./scripts/install.sh`

File: archiveServiceLambda/utility.ts [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    return records.filter((record) => {
+        if (ttlsInSeconds.has(record.dynamodb.NewImage?.resourceType.S)) {
+            const documentStatus = record.dynamodb.NewImage.documentStatus.S;
+            if (record.eventName === 'INSERT' && documentStatus === 'AVAILABLE') {
+                // brand new item is inserted
+                return true;
+            }
+
+            if (record.eventName === 'MODIFY' && ['AVAILABLE', 'DELETED'].includes(documentStatus)) {
+                const onlyTtlChange = _.isEqual(
```
Comment (Zambonilli) [inline archiveServiceLambda/utility.ts:58]: can we double check the lodash isEqual's equality behavior here is recursive.

**Review by nirojshrestha019 — APPROVED** [review]

**Pull request closed without merging**